### PR TITLE
Efficient watch resumption beta

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/enhancements/issues/1904

Manual tests (both scalability and non-scalability related) were done and briefly described in https://github.com/kubernetes/enhancements/pull/2538

```release-note
NONE
```

/kind feature
/priority important-soon

### CODE REVIEW

**Comment by wojtek-t** [comment]
With necessary manual tests already performed (upgrade/downgrade/update), scalability), this is now ready for review.

/assign @deads2k @lavalamp

File: staging/src/k8s.io/apiserver/pkg/features/kube_features.go [hunk 1]
```diff
@@ -185,6 +185,6 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	RemoveSelfLink:           {Default: true, PreRelease: featuregate.Beta},
 	SelectorIndex:            {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
 	WarningHeaders:           {Default: true, PreRelease: featuregate.Beta},
-	EfficientWatchResumption: {Default: false, PreRelease: featuregate.Alpha},
+	EfficientWatchResumption: {Default: true, PreRelease: featuregate.Beta},
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/features/kube_features.go:188]: update the comment for the `EfficientWatchResumption` featuregate around like 153

**Comment by deads2k** [comment]
lgtm modulo the comment

File: staging/src/k8s.io/apiserver/pkg/features/kube_features.go (hunk 1 above)
Comment (wojtek-t) [reply staging/src/k8s.io/apiserver/pkg/features/kube_features.go:188]: done

**Pull request merged**